### ISSUE TITLE
Explicit `Rec.` prefix on unsupported runtime not caught at compile time, may freeze client

### ISSUE THREAD

**Issue by dtkb** [microsoft/AL#6426]
**Describe the bug**
If a page has added explicit `Rec.` qualifiers to fields or properties, these are not supported in runtimes before 6, but there are no warnings at compile-time, so if we backport an extension or just include `Rec.` out of habit to an older runtime, we find only later that it doesn't work, and have to waste our time and disturb users by republishing without `Rec.`

Worse, not all unsupported cases are even reported at runtime; for example, I just found that `Rec.` in a `HideValue` expression just causes the client to lock up and spew a bunch of errors in an infinite loop!

![image](https://user-images.githubusercontent.com/63455896/105180583-aa5a6a80-5b22-11eb-856d-f6ade9d50c27.png)

**To Reproduce**
Here is the page after I _mostly_ fixed it but forgot the `Rec.` in the `HideValue` expression, which caused the inability to open the page as it silently spun an infinite loop as per above.

**Expected behavior**
If compiling against a runtime that doesn't support `Rec.` anywhere, stop the developer with an error, rather than deferring that until runtime and wasting everyone's time. Please?

**5. Versions:**
 - AL Language: 6.1.397164
 - Business Central: Version: GB Business Central 15.5 (Platform 15.0.41893.0 + Application 15.5.41926.0)

**Comment by atoader** [comment]
@dtkb is this still a problem?

**Comment by dtkb** [comment]
Yes, the compiler still doesn't warn that unsupported `Rec.` prefixes will either cause a runtime error or simply crash the client. Same example, redone:

![image](https://user-images.githubusercontent.com/63455896/116980169-d22ea200-acbd-11eb-9d48-858cf2864219.png)

I would expect that
* a CU for such older releases - in this case Version: Platform 15.0.36510.0 + Application 15.0.36649.0 - will stop such code causing crashes and just give a 'proper' runtime error
* the AL compiler would know which combinations of versions and properties do and don't support `Rec.` prefixes and, if unsupported, make that a compile time error or *at least* a warning, so we don't waste time publishing that code to site only to have the client crash.

**Comment by SBalslev** [comment]
Hi! Thank you for this detailed report — the description of the client freeze caused by unsupported Rec. prefixes was very helpful.

Since this was reported, all currently supported Business Central runtimes now support the Rec. prefix. The older runtimes (before 6.0) where this was an issue are no longer in mainstream support.

We're closing this as the scenario is no longer applicable to current versions. If you encounter similar compile-time validation gaps on a current runtime, please don't hesitate to open a new issue.

Thank you for helping us improve the AL developer experience!
